packages/bpk-component-toast/src/BpkToast.js

```diff
@@ -0,0 +1,98 @@
+/*
+ * Backpack - Skyscanner's Design System
+ *
+ * Copyright 2017 Skyscanner Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import PropTypes from 'prop-types';
+import React, { Component } from 'react';
+import BpkCard from 'bpk-component-card';
+import BpkLargeExclamationCircle from 'bpk-component-icon/lg/exclamation-circle';
+import BpkCloseButton from 'bpk-component-close-button';
+import { cssModules } from 'bpk-react-utils';
+
+import STYLES from './bpk-toast.scss';
+
+const getClassName = cssModules(STYLES);
+
+class BpkToast extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = { isOpen: true };
+
+    this.onClose = this.onClose.bind(this);
+    this.setAnimation = this.setAnimation.bind(this);
+  }
+
+  componentWillReceiveProps(nextProps) {
+    if (this.props.message !== nextProps.message || nextProps.isVisible) {
+      this.setState({ isOpen: true });
+    }
+  }
+
+  onClose() {
+    this.setState({ isOpen: false });
+    this.props.onClose();
+  }
+
+  setAnimation() {
+    return this.state.isOpen ? 'bpk-toast-animation-forward' : 'bpk-toast-animation-backward';
+  }
+
+  render() {
+    const classNames = [];
+    classNames.push(getClassName('bpk-toast'));
+    classNames.push(getClassName(this.setAnimation()));
+
```

Comment:
If the consumer provides style (or any other props) these should be applied to the component (usually to the outermost-div)

Reply:
I believe this is done on the next line (https://github.com/Skyscanner/backpack/pull/128/files#diff-1b87f02ec1a0abe7a5a6fd05e0ff09beR60). Let me know if that is what you meant or if I have misunderstood something.

Reply:
That'll do it. Maybe best to check that `this.props.className !== null` too, but if it works without that's fine